diagram-viewer/src/main/java/com/powsybl/diagram/viewer/nad/NetworkAreaDiagramController.java

```diff
@@ -0,0 +1,168 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.diagram.viewer.nad;
+
+import com.google.common.io.ByteStreams;
+import com.powsybl.iidm.network.Container;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Substation;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.nad.NetworkAreaDiagram;
+import javafx.beans.property.StringProperty;
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+import javafx.concurrent.Worker;
+import javafx.fxml.FXML;
+import javafx.scene.control.TextArea;
+import javafx.scene.input.ScrollEvent;
+import javafx.scene.web.WebView;
+import netscape.javascript.JSObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class NetworkAreaDiagramController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkAreaDiagramController.class);
+
+    @FXML
+    public WebView diagramWebView;
+    @FXML
+    public TextArea svgContent;
+    @FXML
+    public TextArea info;
+
+    private String html;
+    private String js;
+
+    private final NetworkAreaDiagramJsHandler jsHandler = new NetworkAreaDiagramJsHandler();
+
+    @FXML
+    private void initialize() throws IOException {
+        // Add Zoom management
+        diagramWebView.addEventFilter(ScrollEvent.SCROLL, (ScrollEvent e) -> {
+            if (e.isControlDown()) {
+                double deltaY = e.getDeltaY();
+                double zoom = diagramWebView.getZoom();
+                if (deltaY < 0) {
+                    zoom /= 1.1;
+                } else if (deltaY > 0) {
+                    zoom *= 1.1;
+                }
+                diagramWebView.setZoom(zoom);
+                e.consume();
+            }
+        });
+
+        // Avoid the useless right click on the image
+        diagramWebView.setContextMenuEnabled(false);
+
+        // Set up the listener on WebView changes
+        // A listener has to be added as loading takes time - execute once the content is successfully loaded
+        diagramWebView.getEngine().getLoadWorker().stateProperty().addListener((observable, oldValue, newValue) -> {
+            if (Worker.State.SUCCEEDED == newValue) {
+                JSObject window = (JSObject) diagramWebView.getEngine().executeScript("window");
+                window.setMember("jsHandler", jsHandler);
+                // For easier debugging, redirect the console.log to the jsHandler
+                diagramWebView.getEngine().executeScript("console.log = function(message) " +
+                        "{ jsHandler.log(message); };");
+            }
+        });
```

Comment:
even if not called at the same place for sld, this can be a method in that upper class